packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,36 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target, relatedTarget }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const oldNotContains = target !== this && !this.contains(relatedTarget as DDSCard);
+    const currentCardIndex = Array.from(this.children).indexOf(target as HTMLElement);
+
+    // keep current page if tabbing back into the carousel after previously moving pages
+    if (currentContains && oldNotContains && currentCardIndex === 0) {
+      (this.children[this.start] as HTMLElement).focus();
+      return;
+    }
+
+    if (currentContains) {
+      // going forwards, change page depending on card index
+      if (currentCardIndex >= this.start + this.pageSize) {
+        this.start = currentCardIndex - (currentCardIndex % this.pageSize);
```

Comment:
Considering the following condition:

* We have many cards in carousel
* `this.start` previously was `2`
* Screen size as large as putting 4 cards in a page

And user keeps hitting tab key to reach to the code here. What should happen?

Reply:
@asudo I see, wasn't aware the `this.start` could be set by the user. Added an additional calculation where it considers the offset to calculate the right index if `this.start` isn't a proper multiple of `this.pageSize`.

Now in the condition you provided `this.start` will become the index `6` instead of `4` as it was before i.e. the next page will start with the proper card.